addons/account/models/account_move.py

```diff
@@ -2918,6 +2918,40 @@ def is_internal_partner(partner):
         move.message_subscribe(list(all_followers_ids))
         return move
 
+    def _post_check_invoice_non_negative(self):
+        """It checks that the total amount of a normal account cannot be less than zero.
+
+        In some accounting localizations this should be adjusted/disabled.
+        """
+        for move in self:
+            if (
+                move.is_invoice(include_receipts=True)
+                and float_compare(move.amount_total, 0.0, precision_rounding=move.currency_id.rounding) < 0
+            ):
+                raise UserError(_("You cannot validate an invoice with a negative total amount. "
+                                  "You should create a credit note instead. Use the action menu to transform it into "
+                                  "a credit note or refund."))
+
+    def _post_check_lock_date(self):
+        """When the accounting date is prior to a lock date, change it automatically upon posting.
+
+        /!\ 'check_move_validity' must be there since the dynamic lines will be recomputed outside the 'onchange'
+        environment.
+
+        In some accounting localizations this should be disabled.
+        """
+        for move in self:
+            affects_tax_report = move._affect_tax_report()
+            lock_dates = move._get_violated_lock_dates(move.date, affects_tax_report)
+            if lock_dates:
+                move.date = move._get_accounting_date(move.invoice_date or move.date, affects_tax_report)
+                move.with_context(check_move_validity=False)._onchange_currency()
+
+    def _post_check_localization(self):
+        """Individual checks on accounting localizations can be placed here.
+        """
+        return
```

Comment:
The name is bad, I doesn't have anything to do with l10n

Reply:
> `_post_check_localization`: we will put our other checks to here, like "for credit note a point to a previous invoice must exists", or "a credit note should point to an invoice and not to other credit note"

That can be done anywhere before the `_post`, no need for a method to do that

Reply:
Ok, here is a situation: Customer installs a module, what auto send out e-mails for the posted invoices. Obviously this module owerwrited the _post() method, just like my hungarian invoice localisation module. But unfortunatelly Odoo will run this e-mail send out _post() method before my localisation module, so how the heck can i do that my hungarina lolasiation module throw an usererror when an invoice is not accepted (like someone forgets a VAT type of TAX from one of the lines), because Odoo already posted it and annoucned the customer via a third party module?

With drawing:
-> original _post() is called
    -> it returns with the posted invoices
-> third party module _post(): accounces the customers that they have invoices posted to them
-> hungarian localisation _post(): throws an UserError because it does not fit the hungarian business logic

Why don't you think, that it is better to implement an empty method (what every accounting localisation module can owerwrite and do their own checks) and that method is called *inside* the original _post() method, so any third party module will get the *really* posted invoices as a result.